src/utils/saga/pager.js

```diff
@@ -0,0 +1,77 @@
+import { call } from 'redux-saga/effects';
+import { counter } from './counter';
+import request from 'utils/request';
+
+/**
+ * Paged Helper
+ * @param object req
+ * @example { cached: rolesCached,
+  pager: {
+    ...pager,
+    store: rolesPager
+  },
+  route: {
+    count: '/roles/count',
+    data: '/roles/'
+  },
+  type: 'Roles' }
+*/
+export function* pager(req) {
+  const {
+    cached,
+    pager: { store },
+    route,
+    type
+  } = req;
+
+  const Pager = {
+    count: !store.count
+      ? yield call(counter, { url: route.count, type })
+      : store.count,
+    limit: !req.pager.limit ? store.limit : req.pager.limit,
```

Comment:
This can be `req.pager.limit || store.limit`, right?

Reply:
fixed